activerecord/lib/active_record/connection_adapters/abstract/schema_creation.rb

```diff
@@ -26,6 +26,8 @@ def visit_AlterTable(o)
           sql << o.foreign_key_drops.map { |fk| visit_DropForeignKey fk }.join(" ")
           sql << o.check_constraint_adds.map { |con| visit_AddCheckConstraint con }.join(" ")
           sql << o.check_constraint_drops.map { |con| visit_DropCheckConstraint con }.join(" ")
+          sql << o.exclusion_constraint_adds.map { |con| visit_AddExclusionConstraint con }.join(" ")
```

Comment:
do you need to check `if supports_exclusion_constraints?` here?

Reply:
No I don't think so, since [`exclusion_constraint_adds`](https://github.com/rails/rails/pull/40224/files#diff-fac852e063eee718c2af44dec6f80b1b46f52f8aae7936e5eea6b2eab08cf2f7R548) should always be defined, and will return an empty array regardless. This follows the pattern already established for things like check constraints.